docs/architecture/maui/mvvm-community-toolkit-features.md

```diff
@@ -0,0 +1,211 @@
+---
+title: MVVM Toolkit Features
+description: Helper components to simplify app development and maintenance
+author: michaelstonis
+no-loc: [MAUI]
+ms.date: 03/10/2023
+---
+
+# MVVM Toolkit Features
+
+[!INCLUDE [download-alert](includes/download-alert.md)]
+
+## MVVM Toolkit
+
+The [Model-View-ViewModel (MVVM)](mvvm.md) pattern is a great structural basis for creating our applications. In this pattern, the ViewModel becomes the backbone of our application as it provides communication to our front-end user interface and backing components. To provide integration with the user interface, we will rely on the ViewModel's properties and commands. As detailed in [Updating views in response to changes in the underlying view model or model](mvvm.md#updating-views-in-response-to-changes-in-the-underlying-view-model-or-model), the `INotifyPropertyChanged` interface on our ViewModel to allows changes to our properties to notify when the value is changed. Implementing all of these features means that our ViewModel can end up becoming very verbose. For example, the following code shows a simple ViewModel with properties that raise changes:
+
+```csharp
+public class SampleViewModel : INotifyPropertyChanged
+{
+    private string _name;
+    private int _value;
+
+    public event PropertyChangedEventHandler PropertyChanged;
+
+    public string Name
+    {
+        get => _name;
+        set => this.SetPropertyValue(ref _name, value);
+    }
+
+    public int Value
+    {
+        get => _value;
+        set => this.SetPropertyValue(ref _value, value);
+    }
+
+    protected bool SetPropertyValue<T>(ref T storageField, T newValue, [CallerMemberName] string propertyName = "")
+    {
+        if (Equals(storageField, newValue))
+            return false;
+
+        storageField = newValue;
+        this.RaisePropertyChanged(propertyName);
+
+        return true;
+    }
+
+    protected virtual void RaisePropertyChanged([CallerMemberName] string propertyName = "")
+    {
+        PropertyChanged?.Invoke(this, new PropertyChangedEventArgs(propertyName));
+    }
+}
+```
+
+While some optimizations could be made over time, we will still end up with a fairly verbose set of code for defining our ViewModel. This code can be difficult to maintain and becomes error-prone.
+
+The MVVM Toolkit (`CommunityToolkit.Mvvm`) package can be used to help address and simplify these common MVVM patterns. The MVVM Toolkit, along with newer features to the .NET language, allows for simplified logic, easy adoption into a project, and runtime independence. The example below shows the same ViewModel using components that come with the MVVM Toolkit:
```

Comment:
Perhaps we should include a link to the NuGet package.

Reply:
There is a note on line 68 with a link to install instructions. I struggled with where to put this. Maybe I move it up higher. Thoughts?

```
> [!NOTE]
> The MVVM Toolkit is provided with the `CommunityToolkit.Mvvm` package. For information on how to add the package to your project, see [Introduction to the MVVM Toolkit](/dotnet/communitytoolkit/mvvm/) on the Microsoft Developer Center.
```

Reply:
Ah, yes. The first time it's mentioned introduce it as the CommunityToolkit.Mvvm NuGet package, with link. Then after that just as `CommunityToolkit.Mvvm`.

Reply:
I updated the language there. Let me know if that is what you were thinking.